resources/splunk_service.rb

```diff
@@ -0,0 +1,109 @@
+# Cookbook Name:: cerner_splunk_ingredient
+# Resource:: splunk_service
+#
+# Resource for managing Splunk as a system service
+class SplunkService < ChefCompat::Resource
+  include CernerSplunk::PlatformHelpers, CernerSplunk::ResourceHelpers,
+          CernerSplunk::ServiceHelpers, CernerSplunk::RestartHelpers
+
+  resource_name :splunk_service
+
+  property :name, String, name_property: true, desired_state: false, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :user, String
+  property :ulimit, Integer
+
+  default_action :start
+
+  def after_created
+    package_from_name unless property_is_set? :package
+  end
+
+  def service_name
+    @service_name ||= service_names[package][node['os'].to_sym]
+  end
+
+  action_class do
+    def service_action(desired_action)
+      service service_name do
+        provider Chef::Provider::Service::Systemd if systemd_is_init?
+        supports start: true, stop: true, restart: true, status: true
+        action desired_action
+      end
+    end
+
+    def configure_service
+      # Use run_action to execute immediately and avoid a race condition with /etc/init.d/splunk not existing yet.
```

Comment:
Could you go into this? What's the race condition? Does it have to do with the setting of ulimit values?